### PR TITLE
Add/remove solution items based on directory structure

### LINKED ISSUE #5028: Add sln should add solution items to the sln file
When we do a dotnet add <sln> <src/project/project.csproj> we should make sure that we have src as a solution item in the sln file.

The sln should look similiar to this: 

```

Microsoft Visual Studio Solution File, Format Version 12.00
# Visual Studio 14
VisualStudioVersion = 14.0.25420.1
MinimumVisualStudioVersion = 10.0.40219.1
Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "src", "src", "{CE69E261-F9E9-41E9-A2B2-C7EB547D4231}"
EndProject
Project("{2150E333-8FDC-42A3-9474-1A3956D46DE8}") = "Solution Items", "Solution Items", "{05490A5A-1315-4695-A20E-024EEAD4E97D}"
	ProjectSection(SolutionItems) = preProject
		global.json = global.json
	EndProjectSection
EndProject
Project("{8BB2217D-0F2D-49D1-97BC-3654ED321F3B}") = "AspNetCoreWebApp_NoAuth_1", "src\AspNetCoreWebApp_NoAuth_1\AspNetCoreWebApp_NoAuth_1.xproj", "{485189E1-D720-4123-8CD3-5985E6DD0AB3}"
EndProject
Global
	GlobalSection(SolutionConfigurationPlatforms) = preSolution
		Debug|Any CPU = Debug|Any CPU
		Release|Any CPU = Release|Any CPU
	EndGlobalSection
	GlobalSection(ProjectConfigurationPlatforms) = postSolution
		{485189E1-D720-4123-8CD3-5985E6DD0AB3}.Debug|Any CPU.ActiveCfg = Debug|Any CPU
		{485189E1-D720-4123-8CD3-5985E6DD0AB3}.Debug|Any CPU.Build.0 = Debug|Any CPU
		{485189E1-D720-4123-8CD3-5985E6DD0AB3}.Release|Any CPU.ActiveCfg = Release|Any CPU
		{485189E1-D720-4123-8CD3-5985E6DD0AB3}.Release|Any CPU.Build.0 = Release|Any CPU
	EndGlobalSection
	GlobalSection(SolutionProperties) = preSolution
		HideSolutionNode = FALSE
	EndGlobalSection
	GlobalSection(NestedProjects) = preSolution
		{485189E1-D720-4123-8CD3-5985E6DD0AB3} = {CE69E261-F9E9-41E9-A2B2-C7EB547D4231}
	EndGlobalSection
EndGlobal
```

Note the src item in it. Without this, the project will not show properly on VS. Also, if we are already handling this, just close this issue.

### PR DESCRIPTION
**Pull request by jgoshi**
@piotrpMSFT @livarcocc @krwq @jonsequitur 

Fixes #5028

### CODE REVIEW

File: src/dotnet/SlnHelper.cs [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+using Microsoft.DotNet.Cli.Sln.Internal;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.Tools.Common
+{
+    public static class SlnHelper
+    {
+        public static IList<string> GetSolutionFoldersFromProject(SlnProject project)
```
Comment (livarcocc) [inline src/dotnet/SlnHelper.cs]: Should you just make these extensions?

Comment (jgoshi) [inline src/dotnet/SlnHelper.cs]: Sure

File: src/dotnet/SlnFileExtensions.cs [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+using Microsoft.DotNet.Cli.Sln.Internal;
+using System.Collections.Generic;
+using System.IO;
+using System.Linq;
+
+namespace Microsoft.DotNet.Tools.Common
+{
+    public static class SlnFileExtensions
+    {
+        public static IList<string> GetSolutionFoldersFromProject(this SlnFile slnFile, SlnProject project)
```
Comment (livarcocc) [inline src/dotnet/SlnFileExtensions.cs]: I meant as extensions of SlnProject and SlnProjectCollection.

Comment (jgoshi) [inline src/dotnet/SlnFileExtensions.cs]: Ok thanks. I'll create two new files SlnProjectExtensions and SlnProjectCollectionExtensions.

**Comment by jgoshi** [comment]
@srivatsn

**Pull request merged**